File: 06-3-boardofmessages_fl/test/repository/board_repository_delete_board_test.dart
import 'package:boardofmessages_fl/repository/board_repository.dart';
import 'package:test/test.dart';

// run tests, >>dart test -r expanded

void main() {
  final repository = BoardRepository(true);

  setUp(() async {
    await repository.reset();
  });

  test('deleteExistingBoard', () async {
    final response = await repository.deleteBoard(1);
    expect(response.message, equals('Board removed'));
  });

  test('DeleteInexistingBoard', () async {
    expect(
          () async => await repository.deleteBoard(1333),
          throwsA(isA<Exception>()),
    );
  });
}

File: kotlin_projects/06-3-BoardOfMessageApp_kt2/app/src/test/java/com/example/boardofmessageapp_kt2/repository/common/BoardRepositoryDeleteBoardTest.kt
package com.example.boardofmessageapp_kt2.repository.common

import kotlinx.coroutines.runBlocking
import org.junit.Test

import org.junit.Assert.*
import org.junit.Before
import retrofit2.HttpException

class BoardRepositoryDeleteBoardTest {

    val boardRepository = BoardRepository(true)

    @Before
    fun before() {
        runBlocking { boardRepository.reset() }
    }

    @Test
    fun `test Delete Existing Board`() =  runBlocking {
        val response = boardRepository.deleteBoard(1)
        assertEquals("Board removed", response.message)
    }

    @Test(expected = HttpException::class)
    fun `test Delete Inexisting Board`() =  runBlocking {
        val response = boardRepository.deleteBoard(1333)
    }

}